### ISSUE TITLE
Receiver: Logically split router and ingestor mode

### ISSUE THREAD

**Issue by matej-g** [thanos-io/thanos#5643]
### Is your proposal related to a problem?
Currently all of the code for receiver ingestion / routing logically is bunched up in the same file. A lot of the features are enabled based on `if` conditions sprinkled throughout the receiver source file. This makes it hard to follow the code and changes, as we're implementing things that increase the divide between ingestor and router mode.

On top of that, router mode currently runs with multi TSDB and store API instantiated, even though we do not collect metrics on such nodes, nor should they be queryable / expose tenant stats (see https://github.com/thanos-io/thanos/pull/5623)

### Describe the solution you'd like
Although they are the same component, we could treat ingestor / router receivers internally as separate components with better logic separation in the code. At the same time, receiver in router mode should not have the overhead of running store API.

### Describe alternatives you've considered
Exist with both modes intermingled :cry: 

### Additional context
Previous work in https://github.com/thanos-io/thanos/pull/5623, but I abandoned in favor of larger refactor.

**Comment by haanhvu** [comment]
I'll work on this

**Comment by haanhvu** [comment]
still needed

**Comment by Kartik-Garg** [comment]
I have raised a working PR against this issue :)

**Comment by GiedriusS** [comment]
This still needs to be done. Feel free to ask if you have any questions.

**Comment by aditya109** [comment]
Thanks @GiedriusS, will pick this up.
